resources.py

```diff
@@ -1,12 +1,27 @@
 """REST API."""
-
 import json
+import os
+import MySQLdb
+from airflow_connector import AirflowDB
 from file_adapter import FileAdapter
-from filter import filter_releases, sort
-from flask_restful import Resource, reqparse
+from filter_releases import filter_releases
+from filter_releases import sort
+from flask_restful import reqparse
+from flask_restful import Resource
+# from main import airflow_db
+# from main import db_connection
 
 adapter = FileAdapter('fake_data/fake_release_data.json', 'fake_data/fake_task_data.json')
 release_requests = {}
+# initiate a connection with the airflow database
+cloudsql_unix_socket = os.path.join('/cloudsql', 'istio-release-ui:us-central1:prod-airflow-snapshot-sandbox')
+db_connection = MySQLdb.connect(
+    unix_socket=cloudsql_unix_socket,
+    host='35.193.234.53',
```

Comment:
I suppose all these need to be parameterized?

Reply:
+1 That s why those should be in main. I am confused about the Resource class. Why do we have to use function pointers ? Must every Resource object have a constructor without parameters ?

Reply:
I'm not really sure what you mean by function pointers, so it would be great if you could explain that. With regards to the second question, as far as I can tell/have been able to figure out, yes.

Reply:
With regards to @hklai 's question, yes according to the docs I read I believe they do